### ISSUE TITLE
shell: Standalone updater installation has broken

### ISSUE THREAD

**Issue by mistydemeo** [axodotdev/cargo-dist#1443]
A recent change has broken the installation of standalone updaters via the shell installers. The PowerShell installers appear to be unaffected. On affected versions of cargo-dist, the shell installer information will be updated looking like this:

```sh
_updater_name=""
_updater_bin=""
```

Instead of specifying the actual name and binary. This was probably introduced in a recent refactor in 19af85324be25d023b2bac291aa105f081d7da30, which updated the artifact kind we're using to generate release info. This artifact kind doesn't have updater information; it was missed since jinja2 doesn't error out on accessing a missing struct member. The PowerShell installer still uses the other kind of artifact, so it's unaffected.

**Comment by mistydemeo** [comment]
Reviewing this, I see the problem.

The previous version, and what PS1 still uses, acts on `ExecutableZipFragment`s. Those have an `updater` field which is an `Option<UpdaterFragment>`: https://github.com/axodotdev/cargo-dist/blob/2c4b0441688322caedc854849d7f759472a557bf/cargo-dist/src/backend/installer/mod.rs#L105-L106

On the other hand, the new version uses `FetchableArchive`, which is similar in most respects except that its `updater` field is an `Option<FetchableUpdaterIdx>`: https://github.com/axodotdev/cargo-dist/blob/2c4b0441688322caedc854849d7f759472a557bf/cargo-dist/src/platform.rs#L141-L142

In other words, it's no longer a *struct*, but an index to *look up that struct from* and hence which is missing the `id` and `binary` fields we're trying to fetch.

This is fixable, but also a sign that we need to [configure minijinja](https://docs.rs/minijinja/latest/minijinja/struct.Environment.html#method.undefined_behavior) to actually report errors on accessing undefined values instead of allowing this to happen.
